### ISSUE TITLE
[CI] TopHitsTests testSerialization failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#71635]
This reproduced locally for me on 7.12.

**Build scan:**
https://gradle-enterprise.elastic.co/s/illghxm6dtkrs/tests/:server:test/org.elasticsearch.search.aggregations.metrics.TopHitsTests/testSerialization

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.metrics.TopHitsTests.testSerialization" -Dtests.seed=C9E00E5E78A53669 -Dtests.security.manager=true -Dtests.locale=es-EC -Dtests.timezone=Africa/Casablanca -Druntime.java=8`

**Applicable branches:**
7.12

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.aggregations.metrics.TopHitsTests&tests.test=testSerialization

**Failure excerpt:**
```
org.junit.ComparisonFailure: expected:<...re_failure":false},"[UbVBOyBF":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":false},"CyGZNIXnhFOOTsWahqrkDPdwCMW":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":false},"TwhqTsUVml":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":false},"hgSprfWjZQEZ":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":tru]e}},"sort":[{"_scrip...> but was:<...re_failure":false},"[CyGZNIXnhFOOTsWahqrkDPdwCMW":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":false},"TwhqTsUVml":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":false},"hgSprfWjZQEZ":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":true},"UbVBOyBF":{"script":{"source":"foo","lang":"mockscript"},"ignore_failure":fals]e}},"sort":[{"_scrip...>

  at __randomizedtesting.SeedInfo.seed([C9E00E5E78A53669:9D386F165FF70BF5]:0)
  at org.junit.Assert.assertEquals(Assert.java:115)
  at org.junit.Assert.assertEquals(Assert.java:144)
  at org.elasticsearch.search.aggregations.BaseAggregationTestCase.testSerialization(BaseAggregationTestCase.java:149)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

**Comment by nik9000** [comment]
Looks like I have to backport a fix I made in 7.13 to 7.12. Once jenkins likes it I'll merge and close this.

**Comment by nik9000** [comment]
Should be stopped by #71655.
